pkg/trainer/replicas.go

```diff
@@ -412,8 +412,7 @@ func (s *TFReplicaSet) SyncPods() error {
 		}
 
 		// Filter the unactive pods
-		fieldSelector := "status.phase!=" + string(v1.PodFailed) +
-			",deletionTimestamp!=nil"
+		fieldSelector := fmt.Sprintf("status.phase!=%s", string(v1.PodFailed))
```

Comment:
Why do we remove `deletionTimestamp!=nil`?

Reply:
See my main comment.
Kubernetes doesn't seem to support `deletionTimestamp` as a field selector (see @zjj2wry, comment below as well).
This leads to infinite creation of pods since the resulting error is not caught, and thus the code behaves like no pod was found.

I tried on Kubernetes `1.7.7` and `1.9.4` and this field is not valid in both.
You can try it out by doing 
`kubectl get po --field-selector=deletionTimestamp!=nil`

Which should result in:
`field selector "deletionTimestamp!=nil": field label not supported: deletionTimestamp`